stix_shifter/stix_transmission/src/modules/elastic_ecs/api_client.py

```diff
@@ -24,10 +24,16 @@ def __init__(self, connection, configuration):
             self.endpoint = '_search'
 
         if auth:
-            headers['Authorization'] = b"Basic " + base64.b64encode(
-                (auth['username'] + ':' + auth['password']).encode('ascii'))
-        else:
-            url_modifier_function = self.add_endpoint_to_url_http
+            if 'username' in auth and 'password' in auth:
+                headers['Authorization'] = b"Basic " + base64.b64encode(
+                    (auth['username'] + ':' + auth['password']).encode('ascii'))
+            elif 'api_key' in auth and 'id' in auth:
+                headers['Authorization'] = b"ApiKey " + base64.b64encode(
+                    (auth['id'] + ':' + auth['api_key']).encode('ascii'))
+            elif 'access_token' in auth:
+                headers['Authorization'] = "Bearer " + auth['access_token']
+        """else: # for default elastic search installation with http protocol
```

Comment:
Did you mean to have this commented out. This will mean the url_modifier_function is always None.
Likewise with the add_endpoint_to_url_http method. If it's not needed then you may as well remove it.

Reply:
We have added this code to connect to default elasticsearch installation over http. I can remove this code for now.